### DISCUSSION TITLE
Is there any convenient way to update the font in time when setting it with skinmanager?

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#453] [Q&A] [answered]
When you click the qskmenu trigger signal to set the font, the font will not update immediately like changing the skin.
   connect(fontMenu,&QskMenu::triggered,[=](int index)
                {
            changeFont(index);
                });
 void  changeFont(int index) {
        // 定义一个包含不同字体设置的结构体或类
        struct FontSettings {
            QString family;
            int pointSize;
            bool bold;
        };

        // 创建一个包含不同字体设置的数组
        FontSettings fontSettings[4] = {
            {"SimHei", 10, false},
            {"SimSun", 10, false},
            {"Microsoft YaHei", 10, false},
            {"KaiTi", 10, false}
        };

        // 确保 index 在数组范围内
        if (index < 0 || index >= 4) return;

        // 获取选定的字体设置
        const FontSettings& settings = fontSettings[index];

        QFont font;
        font.setFamily(settings.family);
        font.setPointSize(settings.pointSize);
        font.setBold(settings.bold);

        // 遍历所有角色并设置字体
        QskFontRole::Category categories[] = {
            QskFontRole::Caption,
            QskFontRole::Body,
            QskFontRole::Subtitle,
            QskFontRole::Title,
            QskFontRole::Headline,
            QskFontRole::Display
        };
        QskFontRole::Emphasis emphasisLevels[] = {
            QskFontRole::VeryLow,
            QskFontRole::Low,
           QskFontRole:: Normal,
            QskFontRole::High,
            QskFontRole::VeryHigh
        };

        for (QskFontRole::Category category : categories) {
            for(QskFontRole::Emphasis emphasis:emphasisLevels)
            {
                //qskSkinManager->skin()->setFont(QskFontRole(category,emphasis), font);
                auto skin=qskSkinManager->skin();
                skin->setFont(QskFontRole(category,emphasis), font);
        }

**Comment by hestrro** [comment]
I used recursion. Is the QskObjectTree in the library used to solve this kind of problem?

void traverseQQuickItem(QQuickItem *item) {
        if (!item) return;

        // 打印当前项的信息
        qDebug() << "Item:" << item->objectName() << "Type:" << item->metaObject()->className();

        // 遍历子项
        for (QQuickItem *child : item->childItems()) {
            child->update();
            traverseQQuickItem(child);  // 递归调用

        }
    }

**Comment by uwerat** [comment, marked as answer]
Yo can reset all cached information and trigger an update of a QskItem by sending a QEvent::StyleChange event to it.

Note, that calling:

    qskSkinManager()->skinChanged( qskSkinManager()->skin() );

sends such an event to all items.

**Comment by uwerat** [comment]
Maybe related: https://github.com/uwerat/qskinny/issues/389
